core/src/main/java/io/parsingdata/metal/Shorthand.java

```diff
@@ -99,154 +100,215 @@
+    /** @see io.parsingdata.metal.expression.value.bitwise.Or */ public static BinaryValueExpression or(final ValueExpression left, final ValueExpression right) { return new io.parsingdata.metal.expression.value.bitwise.Or(left, right); }
+    /** @see io.parsingdata.metal.expression.value.bitwise.Or */ public static SingleValueExpression or(final SingleValueExpression left, final SingleValueExpression right) { return last(new io.parsingdata.metal.expression.value.bitwise.Or(left, right)); }
+    /** @see io.parsingdata.metal.expression.value.bitwise.Not */ public static UnaryValueExpression not(final ValueExpression operand) { return new io.parsingdata.metal.expression.value.bitwise.Not(operand); }
+    /** @see io.parsingdata.metal.expression.value.bitwise.Not */ public static SingleValueExpression not(final SingleValueExpression operand) { return last(new io.parsingdata.metal.expression.value.bitwise.Not(operand)); }
+    /** @see ShiftLeft */ public static BinaryValueExpression shl(final ValueExpression left, final ValueExpression right) { return new ShiftLeft(left, right); }
+    /** @see ShiftLeft */ public static SingleValueExpression shl(final SingleValueExpression left, final SingleValueExpression right) { return last(new ShiftLeft(left, right)); }
+    /** @see ShiftRight */ public static BinaryValueExpression shr(final ValueExpression left, final ValueExpression right) { return new ShiftRight(left, right); }
+    /** @see ShiftRight */ public static SingleValueExpression shr(final SingleValueExpression left, final SingleValueExpression right) { return last(new ShiftRight(left, right)); }
+
+    /** "CONstant": instantiates {@link Const} with a {@link Value} object using {@code value} and {@code DEFAULT_ENCODING}. */
     public static SingleValueExpression con(final long value) { return con(value, DEFAULT_ENCODING); }
+
+    /** "CONstant": instantiates {@link Const} with a {@link Value} object using {@code value} the provided {@code encoding}. */
     public static SingleValueExpression con(final long value, final Encoding encoding) { return con(ConstantFactory.createFromNumeric(value, encoding)); }
+
+    /** "CONstant": instantiates {@link Const} with a {@link Value} object using {@code value} and {@code DEFAULT_ENCODING}. */
     public static SingleValueExpression con(final String value) { return con(value, DEFAULT_ENCODING); }
+
+    /** "CONstant": instantiates {@link Const} with a {@link Value} object using {@code value} the provided {@code encoding}. */
     public static SingleValueExpression con(final String value, final Encoding encoding) { return con(ConstantFactory.createFromString(value, encoding)); }
+
+    /** "CONstant": instantiates {@link Const} with {@code value}. */
     public static SingleValueExpression con(final Value value) { return new Const(value); }
```

Comment:
Minor: maybe this should be the first `con()` in the list of shorthands. With most (or all) other shorthands here, the 'first in each list' references the class, where all the other eventually delegate to it.

Reply:
Agreed, but to do this everywhere required lots of stuff to be reordered. But good idea!